bson/bsonrw/value_writer.go

```diff
@@ -430,6 +436,9 @@ func (vw *valueWriter) WriteObjectID(oid primitive.ObjectID) error {
 }
 
 func (vw *valueWriter) WriteRegex(pattern string, options string) error {
+	if !isValidCString(pattern) || !isValidCString(options) {
```

Comment:
Does this have to be checked for every string or is regex sufficient? For example, WriteJavascript and WriteString

Reply:
bsonspec.org only uses cstrings in key names and regexes. JavaScript and String values use string, not cstring, which doesn't have this issue.

Reply:
Makes sense. Thanks for explaining!